assets/js/modules/analytics-4/components/util/whenHasChangedConversionEvents.test.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { render } from '../../../../../../tests/js/test-utils';
+import { createTestRegistry } from '../../../../../../tests/js/utils';
+import { MODULES_ANALYTICS_4 } from '../../datastore/constants';
+import whenHasChangedConversionEvents from './whenHasChangedConversionEvents';
+
+function TestComponent() {
+	return <div data-testid="component" />;
+}
+function FakeWidgetNull() {
+	return <div data-testid="widget-null" />;
+}
+
+describe( 'whenHasChangedConversionEvents', () => {
+	let registry;
+	let provideNewEvents;
+	let provideLostEvents;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		const { select, dispatch } = registry;
+		const { getModuleData } = select( MODULES_ANALYTICS_4 );
+		const { receiveModuleData } = dispatch( MODULES_ANALYTICS_4 );
+
+		provideNewEvents = ( ...newEvents ) =>
+			receiveModuleData( {
+				...( getModuleData() || {} ),
+				newEvents,
+			} );
+		provideLostEvents = ( ...lostEvents ) =>
+			receiveModuleData( {
+				...( getModuleData() || {} ),
+				lostEvents,
+			} );
```

Comment:
I thought about extracting this to a module-level `datastore/test-utils` but it seems early given this is the only use-case so far.